libcudacxx/include/cuda/std/detail/libcxx/include/__config

```diff
@@ -109,10 +103,6 @@ extern "C++" {
 #    define _LIBCUDACXX_HAS_STL_LITERALS
 #  endif
 
-#  if defined(_CCCL_COMPILER_GCC) && __cplusplus < 201103L
-#    error "libc++ does not support using GCC with C++03. Please enable C++11"
-#  endif
-
```

Comment:
Remark: this is removing something outside the scope of this PR, but I am also fine with the change.

Reply:
Sorry, you are right, I should have made a separate PR for that.

Reply:
Please do so next time. You can leave it in now.